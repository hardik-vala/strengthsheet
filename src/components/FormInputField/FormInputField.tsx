import { View } from "react-native";
import { Text, TextInput } from "react-native-paper";
import "./FormInputField.css";

interface FormInputFieldProps {
  label: string;
  placeholder: string;
  value: string;
  onChangeText: (text: string) => void;
  error: boolean;
}

export function FormInputField({
  label,
  placeholder,
  value,
  onChangeText,
  error = false,
}: FormInputFieldProps) {
  return (
    <View
      style={{
        flexDirection: "row",
        justifyContent: "space-between",
        padding: 10,
        width: "100%",
      }}
    >
      <View
        style={{
          flexDirection: "column",
          justifyContent: "center",
        }}
      >
        <Text variant="bodyLarge">{label}</Text>
      </View>
      <TextInput
        id="form-input-field"
        style={{
          textAlign: "right",
          height: 30,
          width: 100,
        }}
        mode="flat"
        dense={true}
        defaultValue=""
        placeholder={placeholder}
        value={value}
        onChangeText={onChangeText}
        error={error}
      />
    </View>
  );
}
