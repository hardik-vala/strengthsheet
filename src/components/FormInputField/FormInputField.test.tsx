import { act, fireEvent, render, screen } from "@testing-library/react-native";
import React from "react";

import { FormInputField } from "./FormInputField";

const TEST_FIELD_HISTORY = {
  fieldName: "My field",
  historyRecords: [
    { datetime: new Date("2023-08-28"), fieldValue: "value4" },
    { datetime: new Date("2023-08-27"), fieldValue: "value3" },
    { datetime: new Date("2023-08-26"), fieldValue: "value2" },
    { datetime: new Date("2023-08-25"), fieldValue: "value1" },
  ]
};

describe("FormInputField", () => {
  it("renders correctly", () => {
    render(
      <FormInputField
        label="My field"
        placeholder="foobar"
        value=""
        fieldHistory={TEST_FIELD_HISTORY}
        onChangeText={() => {}}
        error={false}
      />
    );

    expect(screen).toMatchSnapshot();
  });

  it("renders field history", async () => {
    render(
      <FormInputField
        label="My field"
        placeholder="foobar"
        value=""
        fieldHistory={TEST_FIELD_HISTORY}
        onChangeText={() => {}}
        error={false}
      />
    );

    await act(() => {
      fireEvent.press(screen.getByText("History"));
    });

    expect(screen).toMatchSnapshot();
  });
});
