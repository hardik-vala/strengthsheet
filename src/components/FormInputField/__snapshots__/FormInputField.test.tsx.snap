// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInputField renders correctly 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "justifyContent": "space-between",
      "padding": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        [
          {
            "textAlign": "left",
          },
          {
            "color": "rgba(28, 27, 31, 1)",
            "writingDirection": "ltr",
          },
          [
            {
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": "400",
              "letterSpacing": 0.15,
              "lineHeight": 24,
            },
            undefined,
          ],
        ]
      }
    >
      My field
    </Text>
  </View>
  <View
    style={
      [
        {
          "backgroundColor": "rgba(231, 224, 236, 1)",
          "borderTopLeftRadius": 4,
          "borderTopRightRadius": 4,
        },
        {
          "width": 100,
        },
      ]
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(28, 27, 31, 1)",
          "bottom": 0,
          "height": 1,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "transform": [
            {
              "scaleY": 0.5,
            },
          ],
          "zIndex": 1,
        }
      }
      testID="text-input-underline"
    />
    <View
      style={
        [
          {
            "paddingBottom": 0,
            "paddingTop": 0,
          },
          {
            "minHeight": 30,
          },
        ]
      }
    >
      <TextInput
        cursorColor="rgba(103, 80, 164, 1)"
        defaultValue=""
        editable={true}
        id="form-input-field"
        maxFontSizeMultiplier={1.5}
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="foobar"
        placeholderTextColor="rgba(73, 69, 79, 1)"
        selectionColor="rgba(103, 80, 164, 1)"
        style={
          [
            {
              "margin": 0,
            },
            {
              "height": 30,
            },
            {
              "paddingBottom": 0,
              "paddingTop": 0,
            },
            {
              "color": "rgba(73, 69, 79, 1)",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "letterSpacing": 0.15,
              "lineHeight": undefined,
              "paddingLeft": 16,
              "paddingRight": 16,
              "textAlign": "right",
              "textAlignVertical": "center",
            },
            false,
            [
              {},
            ],
            undefined,
          ]
        }
        testID="text-input-flat"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
</View>
`;
